src/Entity/Leader.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Entity;
+
+use App\Entity\Enum\Role;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\Column;
+use Symfony\Component\Uid\UuidV4;
+use App\Repository\LeaderRepository;
+use App\Entity\Core\Traits\TimestampableEntity;
+use App\Entity\Core\Traits\SoftDeleteableEntity;
+
+#[ORM\Entity(repositoryClass: LeaderRepository::class)]
+class Leader extends Person
+{
+    use TimestampableEntity;
+    use SoftDeleteableEntity;
+
+    #[Id]
+    #[Column(type: 'uuid', unique: true)]
+    private UuidV4 $id;
+
+    #[ORM\ManyToOne(inversedBy: 'leaders')]
+    #[ORM\JoinColumn(nullable: false)]
+    private ?RoleInUnit $roleInUnit = null;
```

Comment:
uhm.. forse questa relazione va alla rovescia?
in leader deve essere una collection di role_in_unit, no?